File: SQL/Expressions/SQLFieldExpression.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLFieldExpression
        Inherits SQLExpression

        Private pstrFieldName As String
        Private pobjTable As SQLSelectTable

        Public Sub New()


        End Sub

        Public Sub New(ByVal strFieldName As String)

            Me.Name = strFieldName

        End Sub

        Public Sub New(ByVal objTable As SQLSelectTable, ByVal strFieldName As String)

            Me.Name = strFieldName
            Me.Table = objTable

        End Sub

        Public Property Name() As String
            Get

                Return pstrFieldName

            End Get

            Set(ByVal value As String)

                pstrFieldName = value

            End Set
        End Property

        Public Property Table() As SQLSelectTable
            Get

                Return pobjTable

            End Get

            Set(ByVal Value As SQLSelectTable)

                pobjTable = Value

            End Set
        End Property

        Friend Overrides Function SQL(ByVal eConnectionType As Database.ConnectionType) As String

            Dim strSQL As String = String.Empty

            If Me.Name = String.Empty Then
                Throw New Exceptions.DatabaseObjectsException("Field Name has not been set.")
            End If

            strSQL &= Misc.SQLFieldNameAndTablePrefix(pobjTable, pstrFieldName, eConnectionType)

            Return strSQL

        End Function

    End Class

End Namespace

File: SQL/Expressions/SQLFieldExpression.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLFieldExpression : SQLExpression
	{
		private string pstrFieldName;
		private SQLSelectTable pobjTable;
			
		public SQLFieldExpression()
		{
		}
			
		public SQLFieldExpression(string strFieldName)
		{
			this.Name = strFieldName;
		}
			
		public SQLFieldExpression(SQLSelectTable objTable, string strFieldName)
		{
			this.Name = strFieldName;
			this.Table = objTable;
		}
			
		public string Name
		{
			get
			{
				return pstrFieldName;
			}
				
			set
			{
				pstrFieldName = value;
			}
		}
			
		public SQLSelectTable Table
		{
			get
			{
				return pobjTable;
			}
				
			set
			{
				pobjTable = value;
			}
		}
			
		internal override string SQL(Serializers.Serializer serializer)
		{
			return serializer.SerializeFieldExpression(this);
		}
	}
}
